math/check_amicable_pair.cpp

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2020 @author iamnambiar
+ * 
+ * @file
+ * A C++ Program to check whether a pair of number is amicable pair or not.
+ * 
+ * \brief
+ * Amicable Pair are two different numbers such that sum of the proper divisor 
+ * of each number is equal to the other number.
+ *  
+ * Wikipedia link for Reference: https://en.wikipedia.org/wiki/Amicable_numbers 
+ */
+#include <iostream>
+#include <cassert>
+#include <cmath>
+
+/**
+ * Function to calculate the sum of all the proper divisor 
+ * of an integer.
+ * @param num First number.
+ * @return Sum of the proper divisor of the number.
+ */
+int sum_of_divisor(int num) {
+    // Variable to store the sum of all proper divisors.
+    int sum = 0;
+    // Using sqrt(num) helps to reduce Time complexity by a factor of sqrt(num)
+    for (int div = 2; div <= sqrt(num); ++div) {
```

Comment:
`sqrt` function is an expensive one. can write int he description that the loop limit can also be written as:
`div * div < num` which is a 2 instruction code on most processors.